src/blueprint/from_config.rs

```diff
@@ -207,7 +207,114 @@ fn to_fields(type_of: &config::Type, config: &Config) -> Valid<Vec<blueprint::Fi
       .trace(name)
   })?;
 
-  Ok(fields.into_iter().flatten().collect())
+  let fields = fields.into_iter().flatten().collect::<Vec<_>>();
+
+  validate_fields(&fields)?;
```

Comment:
move the call to `validate_fields` above inside of the `validate_all` closure. Using `?` will short circuit and exit.

Reply:
i don't think we can do that in-place because a given field can rely on the construction of N number of other fields, so the validation must be done after all fields are known: https://github.com/tailcallhq/tailcall/pull/521/files#diff-0934da3dca97565bf71c4fd107919b495a6a971c292009709dfc4756760ac27eR287